specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2020-03-01/iotdps.json

```diff
@@ -1643,6 +1960,165 @@
         "ipMask"
       ]
     },
+    "PrivateLinkResources": {
+      "description": "The available private link resources for a provisioning service",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "The list of available private link resources for a provisioning service",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/GroupIdInformation"
+          }
+        }
+      }
+    },
+    "GroupIdInformation": {
+      "description": "The group information for creating a private endpoint on a provisioning service",
+      "type": "object",
+      "properties": {
+        "id": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The resource identifier."
+        },
+        "name": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The resource name.",
+          "pattern": "^(?![0-9]+$)(?!-)[a-zA-Z0-9-]{2,49}[a-zA-Z0-9]$"
+        },
+        "type": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The resource type."
+        },
+        "properties": {
+          "$ref": "#/definitions/GroupIdInformationProperties"
+        }
+      },
+      "required": [
+        "properties"
+      ]
+    },
+    "GroupIdInformationProperties": {
+      "description": "The properties for a group information object",
+      "type": "object",
+      "properties": {
+        "groupId": {
+          "type": "string",
+          "description": "The group id"
+        },
+        "requiredMembers": {
+          "description": "The required members for a specific group id",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "requiredZoneNames": {
+          "description": "The required DNS zones for a specific group id",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        }
+      }
+    },
+    "PrivateEndpointConnectionsList": {
+      "description": "The list of private endpoint connections for a provisioning service",
+      "type": "array",
+      "items": {
+        "$ref": "#/definitions/PrivateEndpointConnection"
+      }
+    },
+    "PrivateEndpointConnection": {
```

Comment:
There're common models defined for [PrivateLinks](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/common-types/resource-management/v1/privatelinks.json). Suggest to re-use them.

Reply:
Is this required? or optional? The API has been integrated with Networking and UX and I would like to avoid changing them if possible.

Reply:
It appears that only storage and storagesync team are referring to this common model.

Reply:
This is required for all the new references. Otherwise it's very hard to keep the consistency. Please use the common model as suggested above.

Reply:
The feature has been shipped and is GA. We cannot make this change without breaking this API.